assets/app/scripts/services/data.js

```diff
@@ -96,29 +97,41 @@ angular.module('openshiftConsole')
 
 // resource:  API resource (e.g. "pods")
 // context:   API context (e.g. {project: "..."})
-// callback:  function to be called with the list of the requested resource and context,
+// callback:  (optional) function to be called with the list of the requested resource and context,
 //            parameters passed to the callback:
 //            Data:   a Data object containing the (context-qualified) results
 //                    which includes a helper method for returning a map indexed
 //                    by attribute (e.g. data.by('metadata.name'))
-// opts:      options (currently none, placeholder)
+// opts:      errorNotification - will popup an error notification if the API request fails (default true)
+// returns promise that will be resolved or rejected when the list operation is complete
   DataService.prototype.list = function(resource, context, callback, opts) {
     resource = APIService.toResourceGroupVersion(resource);
     var callbacks = this._listCallbacks(resource, context);
-    callbacks.add(callback);
+    if (callback) {
+      callbacks.add(callback);
+    }
+
+    var deferred = $q.defer();
+    var promises = this._listPromises(resource, context);
+    promises.push(deferred);
 
```

Comment:
agree its more readable but thats a significantly bigger refactor across this file to change all those method signatures to one param instead of two, to me its not worth the added regression risk, but I'm open to other opinions @spadgett @liggitt

i'd be ok with just adding a `cached(resource, context)` that does the checks for watchInFlight and resourceVersion, i think that is your biggest concern right? that its not obvious what watchInFlight && resourceVersion is actually checking.